firewood/src/merkle/node.rs

```diff
@@ -421,6 +421,20 @@ impl NodeType {
     }
 }
 
+impl From<NodeType> for Node {
+    fn from(inner: NodeType) -> Self {
+        let mut s = Self {
+            root_hash: OnceLock::new(),
+            is_encoded_longer_than_hash_len: OnceLock::new(),
+            encoded: OnceLock::new(),
+            inner,
+            lazy_dirty: AtomicBool::new(false),
+        };
+        s.rehash();
+        s
+    }
+}
```

Comment:
Constructors `leaf` and `branch` should be used instead of `Node::new` (deleted below), but when decoding an encoded node, we're left with the arbitrary `NodeType` in some places. This is an easy way to convert between types. 

In general, there's a really bad pattern used here that doesn't work that well with Rust. Whatever needs to do the locking should wrap the types they need locked instead of having wrapper types with locks. So eventually, `NodeType` will be completely refactored into `Node` and only the `Cache` should manage locks. But that's all for another day.